shard/src/ext/record_cog.py

```diff
@@ -0,0 +1,254 @@
+"""
+Cog for recording a voice channel
+"""
+
+from discord.ext import commands
+import discord
+from discord import AudioSink
+
+import zipfile
+import asyncio
+from io import BytesIO
+from struct import pack
+import base64
+
+
+class AsyncThreadEvent(asyncio.Event):
+    """
+    Custom event class for being able to safely set and clear an
+    asynchronous event from multiple threads.
+
+    Shouldn't need a thread safe clear method as the event should only be
+    referenced from the main thread by the time that that method is called.
+    """
+
+    def set(self):
+        self._loop.call_soon_threadsafe(super().set)
+
+
+class WavFile(AudioSink):
```

Comment:
shouldn't this go in the `discord.py` repo?

Reply:
Well, now that we have our own repo, yeah.